docs/source/csdl-spec.md

```diff
@@ -0,0 +1,630 @@
+---
+title: CSDL Specification
+sidebar_title: CSDL Specification
+description: The Composed Schema Definition Language
+status: Draft
+version: "1.0.0"
+authors: 
+  - Jake Dawkins <jake@apollographql.com>
+  - Trevor Scheer <trevor@apollographql.com>
+  - Ashi Krishnan <ashi@apollographql.com>
+---
+
+## Introduction: What is CSDL?
+
+The Composed Schema Definition Language (CSDL) provides a way to describe a GraphQL Schema which is composed from one or more other GraphQL schemas.
+
+The CSDL is intended to be a single artifact which can power a graph router. For example, the CSDL replaces the service list configuration of Apollo Gateway.
+
+CSDL is a subset of the [GraphQL Schema Definition Language](https://spec.graphql.org/). It removes certain SDL features (for instance, type extensions are not permitted) while also requiring the definition of certain types and providing a suite of directives to describe graph topology. In particular, a CSDL can:
+- define [subgraphs](#subgraph) and bind them to [endpoints](#endpoint) (with the required [`csdl_Graph`](#csdl_Graph) type and [`@csdl_endpoint`](#csdl_endpoint) directive)
+- assign fields to subgraphs (with [`@csdl_resolve`](#resolve))
+- declare additional data required and provided by subgraph field resolvers (with [`@csdl_key`](#key), and [`@csdl_resolve`](#csdl_resolve))
```

Comment:
I'll leave this comment when I get there below, but I think that `@csdl_key` is misleading as a term that bundles both the `@key` for a field's requirements and a field's `@requires` selection set.

Reply:
Do you mean `@csdl_resolve`?